utils/src/main/java/uk/gov/pay/commons/utils/xray/XRaySessionProfiler.java

```diff
@@ -79,6 +59,51 @@ public Object profileExecutionOfQuery(DatabaseQuery databaseQuery, Record record
         }
     }
 
+    private String getDatabaseHostnameAndMetadata(DatabaseQuery databaseQuery, AbstractSession abstractSession, HashMap<String, Object> databaseMetadata) {
+        String hostname = "database";
+
+        Accessor accessor = abstractSession.getAccessor();
+        if (accessor == null) {
+            return hostname;
+        }
+        Connection connection = accessor.getConnection();
+        if (connection == null) {
+            return hostname;
+        }
+        updateMetadata(databaseQuery, databaseMetadata, connection);
+
+        return getDatabaseHostName(connection);
+    }
+
+    private void updateMetadata(DatabaseQuery databaseQuery, HashMap<String, Object> databaseMetadata, Connection connection) {
+        try {
+            DatabaseMetaData metadata = connection.getMetaData();
+            databaseMetadata.put("url", metadata.getURL());
+            databaseMetadata.put("user", metadata.getUserName());
+            databaseMetadata.put("driver_version", metadata.getDriverVersion());
+            databaseMetadata.put("database_type", metadata.getDatabaseProductName());
+            databaseMetadata.put("database_version", metadata.getDatabaseProductVersion());
+            databaseMetadata.put("preparation", databaseQuery.isCallQuery() ? "call" : "statement");
+            databaseMetadata.put("sanitized_query", StringUtils.isEmpty(databaseQuery.getSQLString()) ? "" : databaseQuery.getSQLString());
+        } catch (SQLException exception) {
+            logger.warn("Error getting database connection details.");
+        }
+    }
+
+    private String getDatabaseHostName(Connection connection) {
+        String hostname = "database";
+        try {
+            DatabaseMetaData metadata = connection.getMetaData();
+            hostname = new URI((new URI(metadata.getURL())).getSchemeSpecificPart()).getHost();
+            return connection.getCatalog() + "@" + hostname;
+        } catch (URISyntaxException exception) {
+            logger.warn("Error parsing database host name.");
+        } catch (SQLException exception) {
+            logger.warn("Error getting database connection details.");
+        }
+        return hostname;
+    }
```

Comment:
Rather than returning a default `"database"` value, this method could return an `Optional<String>` and allow the caller to decide what the default is.